agenta-backend/agenta_backend/services/llm_apps_service.py

```diff
@@ -165,28 +165,47 @@ async def run_with_retry(
             return result
         except aiohttp.ClientError as e:
             last_exception = e
-            print(f"Error in evaluation. Retrying in {retry_delay} seconds:", e)
+            error_message = f"HTTP error occurred during request: {str(e)}"
+            logger.error(error_message)
+            await asyncio.sleep(retry_delay)
+            retries += 1
+        except asyncio.TimeoutError as e:
```

Comment:
Why did you add these exceptions? These are already handled in the invoke_app code, which means that they cannot be thrown here.

Reply:
For note, the ClientError was the only one there, because the previous code could have failed in the line
`    async with aiohttp.ClientSession() as client:` before the handling of the exceptions.

Please remove the new exceptions as they are useless